services/headless-lms/chatbot/src/llm_utils.rs

```diff
@@ -8,24 +18,158 @@ use tracing::{debug, error, instrument, trace, warn};
 // API version for Azure OpenAI calls
 pub const LLM_API_VERSION: &str = "2024-06-01";
 
-/// Role of a message in a conversation
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-pub enum MessageRole {
-    #[serde(rename = "system")]
-    System,
-    #[serde(rename = "user")]
-    User,
-    #[serde(rename = "assistant")]
-    Assistant,
-}
-
 /// Common message structure used for LLM API requests
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct APIMessage {
     pub role: MessageRole,
+    #[serde(flatten)]
+    pub fields: APIMessageKind,
+}
+
+// todo should be a method??
```

Comment:
do this here inside `impl APIMessage`, it could be called `to_chatbot_conversation_message(conversation_id, order_number)`